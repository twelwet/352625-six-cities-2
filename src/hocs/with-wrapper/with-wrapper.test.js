import React from "react";
import renderer from "react-test-renderer";
import withWrapper from "./with-wrapper.js";

jest.mock(`react-router-dom`, () => ({
  Link: () => null
}));

jest.mock(`../../components/user/user.jsx`, () => `User`);

const props = {
  wrapperClass: `class`
}

const MockComponent = () => {
  return <div>Mock</div>;
};

const MockComponentWrapped = withWrapper(MockComponent);

it(`withWrapped correctly wrap MockComponent`, () => {

  const tree = renderer
    .create(<MockComponentWrapped {...props} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
