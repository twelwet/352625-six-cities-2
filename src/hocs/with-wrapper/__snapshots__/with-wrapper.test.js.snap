// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withWrapped correctly wrap MockComponent 1`] = `
<div
  className="class"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        />
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <User />
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <div>
    Mock
  </div>
</div>
`;
